ci/common-ci.yml

```diff
@@ -96,3 +96,5 @@ stages:
   stage: test
   trigger:
     strategy: depend
+  variables:
+    PIKA_MPI_ENABLE_POOL: 1
```

Comment:
We have to see if there's anything we can do to not require this. At the moment this is a requirement to not have tests hang...

Reply:
Note, these are the test results without an MPI pool: https://gitlab.com/cscs-ci/ci-testing/webhook-ci/mirrors/4700071344751697/7514005670787789/-/pipelines/1542591315. In other words, the CPU tests are fine (on the 6-rank tests, each rank has 6 threads) but the GPU tests aren't (each rank only has 2 threads on daint-gpu). Should we set `PIKA_MPI_ENABLE_POOL: 1` only for the GPU pipelines so we use both? Should we try to find out if we can fix things to run without a pool (yes, but when)?

Reply:
The pool is enabled for CUDA configurations.